src/lib/components/Navigation/Navigation.tsx

```diff
@@ -0,0 +1,341 @@
+}
+
+const MenuButton = ({ label, onClick }: NavigationMenuButtonProps) => {
+
+  return (
+    <Button appearance="base" className="has-icon is-dark" onClick={onClick}>
+      {label}
+    </Button>
+  )
+}
+
+export interface NavigationNavProps {
+  children: ReactNode;
+  className?: string;
+  isCollapsed: boolean;
+}
+
+const Nav = ({ children, className, isCollapsed }: NavigationNavProps) => {
+
+  return (
+    <nav aria-label="main navigation" className={classNames("l-navigation is-maas", className, { "is-collapsed": isCollapsed, "is-pinned": !isCollapsed})}>
+      {children}
+    </nav>
+  )
+}
+
+export interface NavigationDrawerProps {
+  children: ReactNode;
+}
+
+const Drawer = ({ children }: NavigationDrawerProps) => {
+
+  return (
+    <div className="l-navigation__drawer">
+      <div className="p-panel is-dark">
+        {children}
+      </div>
+    </div>
+  )
+}
+
+export interface NavigationControlsProps {
+  children: ReactNode;
+}
+
+const Controls = ({ children }: NavigationControlsProps) => {
+
+  return (
+    <div className="l-navigation__controls">
+      {children}
+    </div>
+  )
+}
+
+export interface NavigationCollapseToggleProps {
+  isCollapsed: boolean;
+  setIsCollapsed: (isCollapsed: boolean) => void;
+  className?: string;
+}
+
+const CollapseToggle = ({
+  isCollapsed,
+  setIsCollapsed,
+  className,
+}: NavigationCollapseToggleProps): JSX.Element => {
+  return (
+    // This block keeps giving this error:
+    // "Type 'Element' is not assignable to type 'ReactNode'."
+    // Might be a bug with react-components.
+    <Tooltip
+      message={
+        <>
+          {!isCollapsed ? "collapse" : "expand"}( <code>&#91;</code> )
+        </>
+      }
+      position="right"
+    >
+      <Button
+        appearance="base"
+        aria-label={`${!isCollapsed ? "collapse" : "expand"} main navigation`}
+        className={classNames(
+          "is-dense has-icon is-dark u-no-margin l-navigation-collapse-toggle",
+          className
+        )}
+        onClick={(e) => {
+          setIsCollapsed(!isCollapsed);
+          // Make sure the button does not have focus
+          // .l-navigation remains open with :focus-within
+          e.stopPropagation();
+          e.currentTarget.blur();
+        }}
+      >
+        <VanillaIcon light name="sidebar-toggle" />
+      </Button>
+    </Tooltip>
+  );
+};
+
+export interface NavigationContentProps {
+  children: ReactNode;
+}
+
+const Content = ({ children }: NavigationContentProps) => {
+
+  return (
+    <div className="p-panel__content">
+      <div className="p-side-navigaiton--icons is-dark">
+        {children}
+      </div>
+    </div>
+  )
+}
+
+export interface NavigationListProps {
+  children: ReactNode;
+}
+
+const List = ({ children }: NavigationListProps) => {
+  return (
+    <ul className="p-side-navigation__list">
+      {children}
+    </ul>
+  )
+}
+
+// Groups are kind of weird since they're basically just an item.
```

Comment:
As explained above, it would make more sense if we added a "group" variant to an item.